R/sdf_sql.R

```diff
@@ -78,50 +78,6 @@ df_from_sql <- function(sc, sql) {
 }
 
 df_from_sdf <- function(sc, sdf, take = -1) {
-  schema <- sdf_sql_schema(sdf)
-  fields <- sdf_sql_schema_fields(schema)
-
-  if (take >= 0) {
-    sdf <- invoke(sdf, "limit", as.integer(take))
-  }
-
-  df <- invoke_static(
-    spark_connection(sdf),
-    "sparklyr.Translator",
-    "dfToCols",
-    sdf
-  )
-
-  dfNames <- lapply(fields, function(x) x$name)
-  rows <- if (length(df) > 0) length(df[[1]]) else 0
-
-  # If this is a resultset with no rows...
-  if (rows == 0) {
-    # Remove invalid fields that have zero length
-    fields <- Filter(function(e) nchar(e$name) > 0, fields)
-    dfNames <- Filter(function(e) nchar(e) > 0, dfNames)
-
-    dfEmpty <- lapply(fields, function(field) {
-      sdf_sql_default_type(field)
-    })
-    names(dfEmpty) <- dfNames
-
-    df <- data.frame(dfEmpty, stringsAsFactors = FALSE, check.names = FALSE)
-  } else {
-    stringData <- unlist(df)
-    df <- as.data.frame(seq_len(rows))
-
-    lapply(seq_along(fields), function(col) {
-      df[[dfNames[[col]]]] <<- sdf_sql_columns_typed(
-        col,
-        stringData,
-        fields,
-        rows)
-    })
-
-    df[[1]] <- NULL
-  }
-
-  df
+  sdf_collect(sdf)
```

Comment:
I think the name of the function could just be "compile_package_jars"
On Fri, Aug 5, 2016 at 6:18 PM Javier Luraschi <notifications@github.com>
wrote:

> In R/sdf_sql.R
> <https://github.com/rstudio/sparklyr/pull/150#discussion_r73767484>:
>
> > -    stringData <- unlist(df)
> > -    df <- as.data.frame(seq_len(rows))
> > -
> > -    lapply(seq_along(fields), function(col) {
> > -      df[[dfNames[[col]]]] <<- sdf_sql_columns_typed(
> > -        col,
> > -        stringData,
> > -        fields,
> > -        rows)
> > -    })
> > -
> > -    df[[1]] <- NULL
> > -  }
> > -
> > -  df
> > +  sdf_collect(sdf)
>
> I believe even arrays need to be deserialized, currently, line by line:
> https://github.com/rstudio/sparklyr/blob/master/R/spark_deserialize.R#L113-L115.
> But yeah, a quick check to make sure this does not make things painfully
> slow, would be good.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/rstudio/sparklyr/pull/150/files/edd35f1a850efdea1a6eb76234c6e4e08a4e1765#r73767484>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAGXx88v7QS1EKNWw493wnrjbKO54Yvxks5qc7a8gaJpZM4JeE4J>
> .
>